Source/WebDriver/WebSocketServer.cpp

```diff
@@ -0,0 +1,211 @@
+}
+
+void WebSocketServer::removeStaticConnection(WebSocketMessageHandler::Connection& connection)
+{
+    m_staticConnections.erase(std::find(m_staticConnections.begin(), m_staticConnections.end(), connection));
+}
+
+void WebSocketServer::removeConnection(WebSocketMessageHandler::Connection& connection)
+{
+    std::erase_if(m_connectionToSession, [&](const auto& pair) {
+        return pair.first == connection;
+    });
+}
+
+std::optional<RefPtr<Session>> WebSocketServer::getSession(WebSocketMessageHandler::Connection& connection)
+{
+    String sessionId = nullString();
+
+    for (const auto& pair : m_connectionToSession) {
+        if (pair.first == connection) {
+            sessionId = pair.second;
+            break;
+        }
+    }
+
+    if (sessionId.isNull())
+        return std::nullopt;
+
+    auto existingSession = m_service.session();
+    if (!existingSession || !(*existingSession) || (*existingSession)->id() != sessionId)
+        return std::nullopt;
+
+    return existingSession;
+}
+
+std::optional<WebSocketMessageHandler::Connection> WebSocketServer::getConnection(const String& sessionId)
+{
+    for (const auto& pair : m_connectionToSession) {
+        if (pair.second == sessionId)
+            return pair.first;
+    }
+
+    return std::nullopt;
+}
+
+String WebSocketServer::getSessionID(const String& resource)
+{
+    // https://w3c.github.io/webdriver-bidi/#get-a-session-id-for-a-websocket-resource
+
+    constexpr auto sessionPrefix = "/session/"_s;
+    if (!resource.startsWith(sessionPrefix))
+        return nullString();
+
+    auto sessionId = resource.substring(sessionPrefix.length());
+    auto uuid = WTF::UUID::parse(sessionId);
+    if (!uuid || !uuid->isValid())
+        return nullString();
+
+    return sessionId;
+}
+
+RefPtr<WebSocketListener> WebSocketServer::startListening(String sessionId)
+{
+    // https://w3c.github.io/webdriver-bidi/#start-listening-for-a-websocket-connection
+    // FIXME Add more listeners when start supporting multiple sessions
+    auto listener = m_listener;
+    auto resourceName = getResourceName(sessionId);
+    listener->resources.push_back(resourceName);
+    return listener;
+}
+
+String WebSocketServer::getResourceName(String sessionId)
+{
+    // https://w3c.github.io/webdriver-bidi/#construct-a-websocket-resource-name
+    if (sessionId.isNull())
+        return "/session"_s;
+
+    return makeString("/session/"_s, sessionId);
+}
+
+String WebSocketServer::getWebSocketURL(const RefPtr<WebSocketListener> listener, String sessionId)
+{
+    // https://w3c.github.io/webdriver-bidi/#construct-a-websocket-url
+    // FIXME: Support secure flag
+
+    auto resourceName = getResourceName(sessionId);
+
+    auto host = listener->host;
+    if (host == "all"_s)
+        host = "localhost"_s;
+
+    return makeString("ws://"_s, host, ":"_s, String::number(listener->port), resourceName);
+}
+
+void WebSocketServer::removeResourceForSession(const String& sessionID)
```

Comment:
You are suing sessionId in the other methods, I don't mind which one, but use the same, please.